### PR TITLE
ci: wire merge_group triggers into required-check workflows

### PR DESCRIPTION
**Pull request by eric-wang-1990**
## What

Wires `merge_group` triggers into the required-check workflows the repo's **existing** merge-queue sender does **not** already cover, so all required checks report on the queued merge commit and the merge queue can be enabled without deadlock.

## Context: an integration-test merge-queue sender already exists

`trigger-integration-tests.yml` already implements the merge-queue flow for the **`JDBC Integration Tests`** required check:
- Ordinary PR → `skip-integration-tests-pr` posts a green **stub** for `JDBC Integration Tests`.
- `integration-test` label → previews the real run (dispatched to `databricks-driver-test`, **replay-only**, thrift + sea).
- **Merge queue** → `merge-queue-jdbc` is the real required gate (replay dispatch to driver-test).
- `remove-label-on-new-commit` drops the approval label on new commits (the re-approval security control).

That workflow handles the **integration** gate. It does **not** wire the repo's other required checks for the queue. This PR fills exactly that gap.

## Required checks on `main` and how each is covered in the queue

| Required check | Workflow | Queue coverage |
|---|---|---|
| `JDBC Integration Tests` | `trigger-integration-tests.yml` | ✅ Already handled (pre-existing) — not touched by this PR |
| `unit-tests` (6 variants), formatting, packaging | `prCheck.yml` | **This PR:** `+ merge_group` — runs for real against the merge commit (checkout already falls back to `github.ref_name`/`github.repository`; secret steps gated on `push`, so no secrets in queue) |
| `freeze-check` | `releaseFreeze.yml` | **This PR:** `+ merge_group` + job-level skip — PR-semantic gate, already enforced at PR time; skipped required check counts as pass under classic branch protection |
| `check-next-changelog` | `checkNextChangelog.yml` | **This PR:** same skip-in-queue pattern |

`prIntegrationTests.yml` is intentionally **not** wired for the queue — the integration gate is already `merge-queue-jdbc`; a second local mocked run would be a redundant, non-required duplicate that only adds latency.

## Why the job-level skip is safe here

`main` uses **classic branch protection** (`required_status_checks`, `strict: true`; the two repo rulesets are unrelated). Under classic protection a **skipped required check counts as a pass**, so skipping the two PR-semantic jobs in the queue does not block it. They read `github.event.pull_request.*` (body/diff), which doesn't exist in the `merge_group` context, so they must not execute there.

## Scope / follow-ups (NOT in this PR)

- ❌ Enabling "Require merge queue" on the branch rule.
- ❌ Any change to the existing `trigger-integration-tests.yml` sender.
- ❌ Pre-merge **live** (passthrough) E2E — by design the integration gate is replay-only; live remains manual/nightly in driver-test.

Enabling the queue itself should be paired with the standard guardrails (required approval + dismiss-stale-reviews); note the integration sender already enforces re-approval on new commits via `remove-label-on-new-commit`.

NO_CHANGELOG=true

This pull request and its description were written by Isaac.

### CODE REVIEW

File: .github/workflows/skip-checks-reporter.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# `merge_group` commit (and as a label preview on internal PRs), dispatched by
+# trigger-integration-tests.yml. This workflow only posts the pre-merge placeholder.
+
+on:
+  workflow_run:
+    workflows: ["Trigger Integration Tests"]
+    types: [requested]
+
+jobs:
+  report-skip:
```
[the remark this answers is not in the document]
Comment (eric-wang-1990) [reply .github/workflows/skip-checks-reporter.yml:35]: Leaving the guard as-is — this matches the sibling reference `adbc-drivers/databricks` (the origin of this `skip-checks-reporter.yml` pattern) verbatim.

That repo's reporter uses the identical guard (`if: github.event.workflow_run.event == 'pull_request'`) with the same `trigger-integration-tests.yml` triggers (`[opened, synchronize, reopened, labeled, closed]` + `merge_group`), and its App-posted `C# Integration Tests` / `Rust Integration Tests` stubs are the required contexts on `main`. So the App stub firing on `labeled`/`closed` is an accepted characteristic of the established pattern, not a regression this PR introduces:

- **`labeled`:** the real preview run (dispatched to driver-test) posts the actual `JDBC Integration Tests` result on the same head SHA and supersedes the transient "Skipped" stub. The only window where the stub is the final state is if the dispatched run never posts at all — the same edge case the reference accepts.
- **`closed`:** a no-op stub on a closed PR head; harmless.

The one intentional divergence from ADBC is that this PR drops the inline `github.token` stub job that ADBC keeps: JDBC's `JDBC Integration Tests` check is pinned to the App id in branch protection, so a same-name `github.token` check wouldn't satisfy the gate — App-only via the reporter is required here. Resolving to keep parity with the reference.

**Review by msrathore-db — APPROVED** [review]
